src/applications/financial-status-report/components/otherExpenses/OtherExpensesChecklist.jsx

```diff
@@ -40,19 +39,34 @@ const OtherExpensesChecklist = ({
   };
 
   // Calculate Discretionary income as necessary
-  const updateStreamlinedValues = () => {
+  useEffect(() => {
     if (otherExpenses?.length || !gmtData?.isEligibleForStreamlined) return;
 
-    const calculatedDiscretionaryIncome = calculateDiscretionaryIncome(data);
-    setFormData({
-      ...data,
-      gmtData: {
-        ...gmtData,
-        discretionaryBelow:
-          calculatedDiscretionaryIncome < gmtData?.discretionaryIncomeThreshold,
-      },
+    getMonthlyExpensesAPI(data).then(({ calculatedMonthlyExpenses }) => {
+      try {
+        const { totalMonthlyNetIncome } = getMonthlyIncome(data);
+        const calculatedDiscretionaryIncome =
+          totalMonthlyNetIncome - calculatedMonthlyExpenses;
+
```

Comment:
I'm wondering if instead of using .then it would be beneficial to use async/await
```js
useEffect(() => {
    const calculateExpenses = async () => {
      if (!otherExpenses?.length || !gmtData?.isEligibleForStreamlined) return;

      try {
        const { calculatedMonthlyExpenses } = await getMonthlyExpensesAPI(data);
        const { totalMonthlyNetIncome } = getMonthlyIncome(data);
        const calculatedDiscretionaryIncome =
          totalMonthlyNetIncome - calculatedMonthlyExpenses;

        setFormData({
          ...data,
          gmtData: {
            ...gmtData,
            discretionaryBelow:
              calculatedDiscretionaryIncome <
              gmtData?.discretionaryIncomeThreshold,
          },
        });
      } catch (error) {
        Sentry.withScope(scope => {
          scope.setExtra('error', error);
          Sentry.captureMessage(
            `calculate_monthly_expenses failed in OtherExpensesSummary: ${error}`,
          );
        });
      }
    };

    calculateExpenses();
  }, [data, gmtData, setFormData]);
  ```

Reply:
I added `data, gmtData and setFormData` to the useEffect dependency but I haven't tested it in the browser to see if it triggers an infinite loop ⚠️

Reply:
Could you elaborate as to why you think it'd be beneficial? Other than just a stylistic change?

Reply:
@amponce I think in this case the `.then` functions similar to the `await` since both will execute after the promise is resolved. Just running it on page load will help prevent us from spamming the BE with requests with how often the forms system updates `data`, and if any of 
the other expenses are edit, they navigate away and back to this page, triggering the calculation on load again, so it should be pretty safe!